tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc

```diff
@@ -465,47 +466,73 @@ TRTEngineOp::EngineCtxPair& TRTEngineOp::GetEngine(int batch_size,
                                      PluginFactoryTensorRT::GetInstance()));
     auto raw_static_engine = static_engine.get();
     const auto max_batch_size = raw_static_engine->getMaxBatchSize();
-    engine_map_[max_batch_size] = {
-        std::move(static_engine),
-        TrtUniquePtrType<nvinfer1::IExecutionContext>(
-            raw_static_engine->createExecutionContext())};
+    // Static engine will have max_batch_size for batch size so that all inputs
+    // will map to this single engine.
+    for (int i = 0; i < input_shapes.size(); i++) {
+      // TODO(tmorris): will all inputs have batch size as first dimension??
+      input_shapes[i].set_dim(0, max_batch_size);
+    }
+    lru_cache_.emplace(
+        input_shapes,
```

Comment:
Here we assume `input_shapes` matches the input shapes of the engine, please help to add a comment for that.